### PR TITLE
EOS:16994 - Add error handling in s3_setup script

### PR DESCRIPTION
**Pull request by amitwac2608**
Signed-off-by: Amit Kumar <amit.kumar@seagate.com>

### CODE REVIEW

File: scripts/provisioning/s3_setup [hunk 1]
```diff
@@ -177,9 +185,22 @@ configure_openldap()
     /opt/seagate/cortx/s3/install/ldap/setup_ldap.sh --confurl "$confstore_config_url" --defaultpasswd --skipssl
     if [ $? -ne 0 ]
     then
-        echo "ERROR: failed to configure openldap, exiting."
+        echo "ERROR: Failed to configure openldap."
+        echo "ERROR: Please re-run 's3_setup config' again."
```
Comment (bkirunge7) [inline scripts/provisioning/s3_setup:189]: last statement can be followed, we can remove this message

File: scripts/provisioning/s3_setup [hunk 2]
```diff
@@ -195,27 +216,21 @@ configure_openldap()
         /opt/seagate/cortx/s3/install/ldap/replication/setupReplicationScript.sh -h hosts_list_file.txt -p "$rootdnpasswd"
         if [ $? -ne 0 ]
         then
-            echo "ERROR: failed to configure ldap replication, exiting."
+            echo "ERROR: Failed to configure ldap replication, exiting."
+            echo "ERROR: You may want to enable openldap logs, and check what exactly failed."
```
Comment (bkirunge7) [inline scripts/provisioning/s3_setup:220]: this statement not required as slapd logs are enabled by default above

File: scripts/provisioning/s3_setup [hunk 3]
```diff
@@ -195,27 +216,21 @@ configure_openldap()
...
-    fi
-    systemctl restart s3authserver
-    systemctl restart rsyslog
+    echo "INFO: Successfully configured openldap on the node."
 }
 
 add_s3backgrounddelete_rabbitmq_user()
 {
-    users=`rabbitmqctl list_users`
+    users=$(rabbitmqctl list_users)
```
Comment (bkirunge7) [inline scripts/provisioning/s3_setup:233]: what if users is empty

File: scripts/provisioning/s3_setup [hunk 4]
```diff
@@ -324,7 +355,7 @@ then
 
     cmd_out=$(ldapsearch -b "o=s3-background-delete-svc,ou=accounts,dc=s3,dc=seagate,dc=com" -x -w $ldappasswd -D "cn=sgiamadmin,dc=seagate,dc=com" -H ldap://) || echo ""
     if [[ $cmd_out == *"No such object"* ]];then
-        echo "ERROR: failed to find s3background delete account, exit."
+        echo "ERROR: failed to find s3background delete account, sanity test failed, exiting."
```
Comment (bkirunge7) [inline scripts/provisioning/s3_setup:358]: is it sanity test ?

File: scripts/provisioning/s3_setup (hunk 3 above)
Comment (amitwac2608) [reply scripts/provisioning/s3_setup:233]: Incase "cortx-s3" user does not exist, or "users" is empty, then we create "cortx-s3" user (Most common scenario)

But if the "users" is not empty and "cortx-s3" user already exists, we first delete "cortx-s3" user, and then go ahead and create "cortx-s3" user again.

File: scripts/provisioning/s3_setup (hunk 4 above)
Comment (amitwac2608) [reply scripts/provisioning/s3_setup:356]: For time being we have kept above "ldapsearch" as sanity test, as our santiy test script: "s3-santiy-test.sh" needs to be fixed.

File: scripts/provisioning/s3_setup (hunk 1 above)
Comment (amitwac2608) [reply scripts/provisioning/s3_setup]: done

File: scripts/provisioning/s3_setup (hunk 2 above)
Comment (amitwac2608) [reply scripts/provisioning/s3_setup]: done

File: scripts/ldap/setup_ldap.sh [hunk 5]
```diff
@@ -189,12 +189,9 @@ ldapmodify -Y EXTERNAL -H ldapi:/// -w $ROOTDNPASSWORD -f $INSTALLDIR/resultssiz
 systemctl enable slapd
 systemctl restart slapd
 
-echo "Encrypting Authserver LDAP password"
+echo "Encrypting Authserver LDAP password.."
 /opt/seagate/cortx/auth/scripts/enc_ldap_passwd_in_cfg.sh -l $LDAPADMINPASS -p /opt/seagate/cortx/auth/resources/authserver.properties
 
-echo "Restart S3authserver"
-systemctl restart s3authserver
```
Comment (AjinkyaDhumal) [inline scripts/ldap/setup_ldap.sh:196]: Lets not remove this for sometime till HA component is ready to manage the lifecycles.

**Review by bkirunge7 — APPROVED** [review]

File: scripts/ldap/setup_ldap.sh (hunk 5 above)
Comment (amitwac2608) [reply scripts/ldap/setup_ldap.sh:196]: ok

Comment (amitwac2608) [reply scripts/ldap/setup_ldap.sh:196]: done

**Review by AjinkyaDhumal — APPROVED** [review]

**Review by bkirunge7 — APPROVED** [review]

**Review by AjinkyaDhumal — APPROVED** [review]

**Comment by amitwac2608** [comment]
"premerge-test" test failed with below error:
17:00:13  Test case [s3cmd can not upload 18MBfile file] - Execution time - 1 min 3.244 seconds
17:00:13  Command has failed as expected.
17:00:13  Traceback (most recent call last):
17:00:13    F**ile "negative_spec.py", line 370, in <module>
17:00:13      command_error_should_have("ServiceUnavailable")
17:00:13    File "/var/jenkins/workspace/Cortx-PR-Build/Component-Test/S3Server-PreMerge-Test/st/clitests/framework.py", line 178, in command_error_should_have
17:00:13      assert msg in self.status.stderr
17:00:13  AssertionError**

**The above error is not related to my changes, as my changes are in s3_setup and related provisioner script.**

I re-triggered the failed job, Job: http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Component-Test/job/S3Server-PreMerge-Test/104/

which **PASSED**

**Pull request merged**